.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,83 @@
+name: Deploy CCCL pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+
+      # Build helper image for Thrust/CUB
+      - name: Build helper image
+        run: |
+          bash ./docs/make_env.bash "cccl:docs"
+
+      # Build top level docs for CCCL
+      - name: Build landing page
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./docs/jekyll
+          destination: ./_site
+
+      # CUB
+      - name: Build CUB docs
+        run: |
+          bash ./docs/build_docs.bash "cccl:docs" /cccl/cub/docs/gen_docs.bash
+          sudo mkdir -p ./_site/cub

```

Comment:
I don't know why, but I couldn't figure out the permissions for the output directory. We need `sudo` to allow writing into it.